tasks/system/RedHat-7/install_docker.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Remove docker
+  package:
+    name: "{{ packages }}"
+    state: absent
+  vars:
+    packages:
+    - docker
+  register: remove_packages
+  retries: 10
+  delay: 30
+  until: remove_packages is success
+
+- name: Add Docker GPG Key
+  rpm_key:
+    key: https://download.docker.com/linux/centos/gpg
+    state: present
+
+- name: Add docker repository
+  yum_repository:
+    name: "{{ docker_version_map[docker_version]['name'] }}"
+    description: "Docker repository"
+    file: docker-ce
+    baseurl: https://download.docker.com/linux/centos/7/x86_64/stable
+    enabled: yes
+    gpgcheck: no
+  when: docker_version == '18.09'
```

Comment:
Maybe we can add a field to the `docker_version_map` for RHEL `add_docker_repo` or so that is used here as a `when: add_docker_repo is defined`. This way we would only need to maintain that logic in the version map instead of going back to this task for new versions that get added later... 
What do you think?